Version 0.5.0: Cargo Tries to link to SspiCli.dll instead of Secur32.dll when using GetUserNameExA()

When trying to use `GetUserNameExA()`, `cargo build` tries to link to `SspiCli.dl` using non-existing file `SspiCli.lib` instead of `Secur32.dll` using `Secur32.lib`, and build fails. I have the Windows 10 SDK installed and Secur32.lib is included in the SDK, so linking with that would succeed.

# Steps to reproduce
Create a new project with the following:
## Cargo.toml
```
[package]
name = "seclib"
version = "0.1.0"
edition = "2018"

[dependencies]
windows = "0.5.0"

[build-dependencies]
windows = "0.5.0"

```

## build.rs
```rust
fn main() {
    windows::build!(
        windows::win32::windows_programming::GetUserNameExA
    );
  }
  
```

## src\main.rs
```rust
#![allow(dead_code)]
#![allow(unused_imports)]

mod bindings {
    ::windows::include_bindings!();
}


use bindings::windows::win32::windows_programming::GetUserNameExA;

fn main() {
    println!("Hello, world!");
}
```

# Expected results
project is linked with `secur32.dll` library and is linked successfully when executing `cargo build`

# Actual results
cargo build tries to link to `SspiCli.dl` using non-existing file `SspiCli.lib`, and build fails
```
PS C:\Users\vegard\source\repos\seclib> cargo build
   Compiling seclib v0.1.0 (C:\Users\vegard\source\repos\seclib)
error: linking with `link.exe` failed: exit code: 1181
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.28.29910\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" 
"/NXCOMPAT" "/LIBPATH:C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\seclib.1hpx7l8yy80fv79s.rcgu.o" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\seclib.1w4z1sgbrtqnze7z.rcgu.o" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\seclib.2fs4c3n89fhjdvfv.rcgu.o" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\seclib.2qslwrgqz64qcgqv.rcgu.o" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\seclib.3reqzfqie3wgb0fq.rcgu.o" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\seclib.411tlv2x1c6j2qno.rcgu.o" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\seclib.k435cl9zshn99or.rcgu.o" "/OUT:C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\seclib.exe" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\seclib.14z1cf9we081h8dx.rcgu.o" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis" "/LIBPATH:C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps" "/LIBPATH:C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "SspiCli.lib" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\libwindows-8bd0580c6cfa550c.rlib" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\libconst_sha1-b23143bff14a823b.rlib" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\libwindows_gen-07bcc77ef0066c8f.rlib" "C:\\Users\\vegard\\source\\repos\\seclib\\target\\debug\\deps\\libsquote-7f66eb9f64fa600d.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-0087fff1d6c84113.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-e9a9adcb7b282f86.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-c17bc77d789f8359.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-1f739c53c46039b3.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-b12f9afb50af9e82.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-4813d8391ae0f1e1.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-a6dfd598b03abbd4.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-8e04de9ca6363561.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-05126e82a7fe403c.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-7f3e1ef8417012fa.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-df80c014103f27c5.rlib" "C:\\Users\\vegard\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-f404d34c325d6d79.rlib" "OLEAUT32.lib" "OLEAUT32.lib" "OLEAUT32.lib" "OLEAUT32.lib" "OLEAUT32.lib" "OLE32.lib" "OLE32.lib" "OLE32.lib" "OLE32.lib" "OLE32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "KERNEL32.lib" "advapi32.lib" "ws2_32.lib" "userenv.lib" "msvcrt.lib"
  = note: LINK : fatal error LNK1181: cannot open input file 'SspiCli.lib'


error: aborting due to previous error

error: could not compile `seclib`

To learn more, run the command again with --verbose.
```

--- a/crates/gen/src/types/function.rs
+++ b/crates/gen/src/types/function.rs
@@ -46,7 +46,7 @@ impl Function {
         let mut link = self.0.impl_map().expect("Function").scope().name();
 
         // TODO: workaround for https://github.com/microsoft/windows-rs/issues/463
-        if link.contains("-ms-win-") || link == "D3DCOMPILER_47" {
+        if link.contains("-ms-win-") || link == "D3DCOMPILER_47" || link == "SspiCli" {
             link = "onecoreuap";
         }
 

--- a/crates/macros/src/build_limits.rs
+++ b/crates/macros/src/build_limits.rs
@@ -106,14 +106,14 @@ fn use_tree_to_namespace_types(use_tree: &syn::UseTree) -> syn::parse::Result<Na
                 recurse(reader, &*p.tree, current)
             }
             syn::UseTree::Glob(g) => {
-                let namespace = find_namespace(reader, &current.clone(), g.span())?;
+                let namespace = find_namespace(reader, &current, g.span())?;
                 Ok(NamespaceTypes {
                     namespace,
                     limit: TypeLimit::All,
                 })
             }
             syn::UseTree::Group(g) => {
-                let namespace = find_namespace(reader, &current.clone(), g.span())?;
+                let namespace = find_namespace(reader, &current, g.span())?;
 
                 let mut types = Vec::with_capacity(g.items.len());
                 for tree in &g.items {
@@ -136,7 +136,7 @@ fn use_tree_to_namespace_types(use_tree: &syn::UseTree) -> syn::parse::Result<Na
                 })
             }
             syn::UseTree::Name(n) => {
-                let namespace = find_namespace(reader, &current.clone(), n.span())?;
+                let namespace = find_namespace(reader, &current, n.span())?;
                 let name = n.ident.to_string();
                 Ok(NamespaceTypes {
                     namespace,

--- a/crates/macros/src/implement.rs
+++ b/crates/macros/src/implement.rs
@@ -5,15 +5,19 @@ use squote::{format_ident, quote, Literal, TokenStream};
 struct Implements(Vec<gen::ElementType>);
 
 impl syn::parse::Parse for Implements {
-    fn parse(inner_type: syn::parse::ParseStream) -> syn::parse::Result<Self> {
+    fn parse(input: syn::parse::ParseStream) -> syn::parse::Result<Self> {
         let mut types = Vec::new();
         let reader = gen::TypeReader::get();
 
         loop {
-            use_tree_to_types(reader, &inner_type.parse::<ImplementTree>()?, &mut types)?;
-
-            if inner_type.parse::<syn::Token!(,)>().is_err() {
+            if input.is_empty() {
                 break;
+            }
+
+            use_tree_to_types(reader, &input.parse::<ImplementTree>()?, &mut types)?;
+
+            if !input.is_empty() {
+                input.parse::<syn::Token![,]>()?;
             }
         }
 

--- a/crates/macros/src/implement_tree.rs
+++ b/crates/macros/src/implement_tree.rs
@@ -12,7 +12,6 @@ pub enum ImplementTree {
 
 pub struct ImplementPath {
     pub ident: Ident,
-    pub colon2_token: Token![::],
     pub tree: Box<ImplementTree>,
 }
 
@@ -32,9 +31,9 @@ impl Parse for ImplementTree {
         if lookahead.peek(Ident) {
             let ident = input.call(Ident::parse_any)?;
             if input.peek(Token![::]) {
+                input.parse::<syn::Token![::]>()?;
                 Ok(ImplementTree::Path(ImplementPath {
                     ident,
-                    colon2_token: input.parse()?,
                     tree: Box::new(input.parse()?),
                 }))
             } else {

